### PR TITLE
Feature(s): Add scoreboard as data source + sunmoon

### PR DESCRIPTION
**Pull request by RayDeeUx**
also for some reason `RefreshSeconds` -> `refreshSeconds` per variable naming conventions

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt [hunk 1]
```diff
@@ -16,13 +18,31 @@ class InGameDateDisplay {
...
     }
 
     private fun checkDate() {
-        display = SkyBlockTime.now().formatted()
+        val date = SkyBlockTime.now()
+        var theBaseString = ""
+        if (config.useScoreboard) {
+            val list = ScoreboardData.sidebarLinesFormatted //we need this to grab the moon/sun symbol
+            val year = "Year ${date.year}"
+            val monthAndDate = list.find{ (it.contains("Winter ") || it.contains("Spring ") || it.contains("Summer ") || it.contains("Autumn ")) && (it.contains("st") || it.contains("nd") || it.contains("rd") || it.contains("th")) } ?: "??"
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt]: Please use mixins for stuff like `list.find{ (it.contains("Winter ") || it.contains("Spring ") || it.contains("Summer ") || it.contains("Autumn "))`

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Please make this code less dense in complexity. Either refactor into multiple functions or use mixins for the checks or do both

File: src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt (hunk 1 above)
Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt:34]: hanni what the heck is a mixin

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt]: I confused mixin with regex in that context. I'm sorry

Comment (RayDeeUx) [reply src/main/java/at/hannibal2/skyhanni/features/misc/InGameDateDisplay.kt]: it has been done.

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**